pulp_container/app/registry_api.py

```diff
@@ -345,36 +349,132 @@ def get(self, request):
         return Response(data={})
 
 
-class CatalogView(ContainerRegistryApiMixin, APIView):
+class ContainerCatalogSerializer(ModelSerializer):
+    """
+    Serializer for Distributions in the _catalog endpoint of the registry.
+    """
+
+    class Meta:
+        model = models.ContainerDistribution
+        fields = ["base_path"]
+
+
+class ContainerCatalogPagination(BasePagination):
+    """
+    Pagination class to paginate repositories by names according to the registry api specification.
+    """
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """
+        Analyse the pagination parameters and prepare the queryset.
+        """
+        try:
+            self.n = int(request.query_params.get("n"))
+        except Exception:
+            self.n = self.page_size
+        last = request.query_params.get("last")
+        self.url = request.build_absolute_uri()
+
+        if last:
+            queryset = queryset.filter(base_path__gt=last)
+        return queryset.order_by("base_path")[: self.n]
```

Comment:
we can return here qs.value_list('base_path', flat=true) so then later you can create a list and not iterate trough qs.

Reply:
Hmm i turned this endpoint into a ListViewset, and it now uses a serializer. I think this will not work well with `value_list`.
Maybe `only("base_path")` can be an improvement still.